### PR TITLE
docs(switch): implement and document variables and attributes

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

`<hx-switch>` : Implement and document variables and attributes

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2128

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [ ] Did you assign reviewers?
- [ ] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/scss/components/switch/_index.scss [hunk 1]
```diff
@@ -39,10 +39,12 @@ hx-switch-control {
 hx-switch-control > input[type="checkbox"] ~ label {
   > hx-switch {
     background-color: $gray-0;
+    background-color: var(--hxSwitchControl-switch-bgcolor, $gray-0);
     border: 0.125rem solid $cyan-900;
+    border: 0.125rem solid var(--hxSwitchControl-switch-border-color, $cyan-900);
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:44]: Please use updated colors to replace `$cyan-900`.

File: src/scss/components/switch/_index.scss [hunk 2]
```diff
@@ -39,10 +39,12 @@ hx-switch-control {
 hx-switch-control > input[type="checkbox"] ~ label {
   > hx-switch {
     background-color: $gray-0;
+    background-color: var(--hxSwitchControl-switch-bgcolor, $gray-0);
     border: 0.125rem solid $cyan-900;
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:43]: Please use updated colors to replace `$cyan-900`.

File: src/scss/components/switch/_index.scss [hunk 3]
```diff
@@ -51,6 +53,7 @@ hx-switch-control > input[type="checkbox"] ~ label {
     text-transform: uppercase;
     margin-left: 2rem;
     color: $cyan-900;
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:55]: Please use updated colors to replace `$cyan-900`.

File: src/scss/components/switch/_index.scss [hunk 4]
```diff
@@ -51,6 +53,7 @@ hx-switch-control > input[type="checkbox"] ~ label {
...
+    color: var(--hxSwitchControl-switch-offlabel-color, $cyan-900);
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:56]: Please use updated colors to replace `$cyan-900`.

File: src/scss/components/switch/_index.scss [hunk 5]
```diff
@@ -62,7 +65,9 @@ hx-switch-control > input[type="checkbox"] ~ label {
 
   > hx-switch::before {
     background-color: $gray-0;
+    background-color: var(--hxSwitchControl-switch-bgcolor, $gray-0);
     border: 0.125rem solid $cyan-900;
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:69]: Please use updated colors to replace `$cyan-900`.

File: src/scss/components/switch/_index.scss [hunk 6]
```diff
@@ -62,7 +65,9 @@ hx-switch-control > input[type="checkbox"] ~ label {
...
+    border: 0.125rem solid var(--hxSwitchControl-switch-border-color, $cyan-900);
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:70]: Please use updated colors to replace `$cyan-900`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 7]
```diff
@@ -9,38 +9,49 @@
 
     &::-ms-check {
       border-color: $cyan-700;
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:11]: Please use updated colors to replace `$cyan-700`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 8]
```diff
@@ -9,38 +9,49 @@
...
+      border-color: var(--hxSwitchControl-focus-border-color, $cyan-700);
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:12]: Please use updated colors to replace `$cyan-700`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 9]
```diff
@@ -9,38 +9,49 @@
...
     }
   } @else if $pseudo-state == hover {
     @if $checkBox-state == unchecked {
       background-color: $gray-0;
+      background-color: var(--hxSwitchControl-hover-unchecked-bgcolor, $gray-0);
       border: 0.125rem solid $cyan-500;
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:18]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 10]
```diff
@@ -9,38 +9,49 @@
...
+      border: 0.125rem solid var(--hxSwitchControl-hover-unchecked-border-color, $cyan-500);
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:19]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 11]
```diff
@@ -9,38 +9,49 @@
...
 
       &::before {
         background-color: $gray-0;
+        background-color: var(--hxSwitchControl-hover-unchecked-before-bgcolor, $gray-0);
         border: 0.125rem solid $cyan-500;
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:24]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 12]
```diff
@@ -9,38 +9,49 @@
...
+        border: 0.125rem solid var(--hxSwitchControl-hover-unchecked-before-border-color, $cyan-500);
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:25]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 13]
```diff
@@ -9,38 +9,49 @@
...
         left: -0.12rem;
       }
 
       &::after {
         color: $cyan-500;
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:30]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 14]
```diff
@@ -9,38 +9,49 @@
...
+        color: var(--hxSwitchControl-hover-unchecked-after-color, $cyan-500);
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:31]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 15]
```diff
@@ -9,38 +9,49 @@
...
       }
     } @else {
       background-color: $cyan-500;
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:34]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 16]
```diff
@@ -9,38 +9,49 @@
...
+      background-color: var(--hxSwitchControl-hover-checked-bgcolor, $cyan-500);
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:35]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 17]
```diff
@@ -56,26 +67,32 @@
 
     &::-ms-check {
       border-color: $red-status-500;
+      border-color: var(--hxSwitchControl-error-focus-border-color, $red-status-500);
     }
   } @else if $pseudo-state == hover {
     &::-ms-check {
       background-color: $red-100;
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:74]: Please use updated colors to replace `$red-XXX` -> `$red-status-XXX`.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss [hunk 18]
```diff
@@ -56,26 +67,32 @@
...
+      background-color: var(--hxSwitchControl-error-hover-bgcolor, $red-100);
```
Comment (100stacks) [inline src/scss/components/switch/mixins/_hxSwitchControl.scss:75]: Please use updated colors to replace `$red-XXX` -> `$red-status-XXX`.

File: src/scss/components/switch/_index.scss [hunk 19]
```diff
@@ -71,6 +76,7 @@ hx-switch-control > input[type="checkbox"] ~ label {
 hx-switch-control > input[type="checkbox"]:checked ~ label {
   > hx-switch {
     background-color: $cyan-900;
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:78]: Please use updated colors to replace `$cyan-XXX`.

File: src/scss/components/switch/_index.scss [hunk 20]
```diff
@@ -71,6 +76,7 @@ hx-switch-control > input[type="checkbox"] ~ label {
...
+    background-color: var(--hxSwitchControl-switch-checked-bgcolor, $cyan-900);
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:79]: Please use updated colors to replace `$cyan-XXX`.

**Review by 100stacks — CHANGES_REQUESTED** [review]
### Please use `v2.0` color variables.

File: src/scss/components/switch/_index.scss (hunk 1 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/_index.scss (hunk 2 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/_index.scss (hunk 3 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/_index.scss (hunk 4 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/_index.scss (hunk 5 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/_index.scss (hunk 6 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 7 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 8 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 9 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 10 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 11 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 12 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 13 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 14 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 15 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 16 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 17 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

File: src/scss/components/switch/_index.scss (hunk 20 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/_index.scss (hunk 19 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/_index.scss]: Changes have been done, please review.

File: src/scss/components/switch/mixins/_hxSwitchControl.scss (hunk 18 above)
Comment (lalithkarikelli) [reply src/scss/components/switch/mixins/_hxSwitchControl.scss]: Changes have been done, please review.

**Review by 100stacks — APPROVED** [review]
### Normally we aren't referencing the `$design-tokens` directly.  This should still work fine.

Good job! Dev LGTM

**Review by badejayayesubabu — APPROVED** [review]
DEV LGTM

**Pull request merged**